Source/WebKit/WebProcess/Plugins/PluginView.cpp

```diff
@@ -850,13 +874,12 @@ void PluginView::invalidateRect(const IntRect& dirtyRect)
         return;
 #endif
 
-    auto* renderer = m_pluginElement->renderer();
-    if (!is<RenderEmbeddedObject>(renderer))
+    auto renderer = checkedRenderer();
```

Comment:
We usually call `checkedX()` / `protectedX()` functions only to avoid local variables. When we have a local variable, we often use something like:
```
CheckedPtr renderer = m_renderer;
```

Not the explicit use of a smart pointer instead of auto. Sadly, checkedRenderer() also does a cast internally.
Maybe we should have a `renderer()` method which does the cast and use this here:
```
CheckedPtr renderer = this->renderer();
```

Reply:
But I want the `dynamicDowncast<RenderEmbeddedObject>()` that `checkedRenderer()` does